store/src/java/com/zimbra/cs/redolog/util/PlaybackUtil.java

```diff
@@ -375,6 +375,7 @@ public static void main(String[] cmdlineargs) throws Throwable {
             player.playback();
         } finally {
             teardown();
+            System.exit(0);
```

Comment:
why is this needed now?

Reply:
Yeah, does this actually stop it from hanging? It seemed like the hangup was something in the teardown(), that calls Zimbra.shutdown(), which just shuts down a lot of threads for the server, but if there was a problem, it's still not going to make it to the next System.exit(0) statement, right?

Reply:
there seems to be something going in that shutdown, Suchit is checking on it.

Reply:
Some new threads which are not exiting are causing the issue which will be refactored and fixed in the respective projects. In the mean time we can go with this change for the tool which has no functional impact.